ush/rocoto/setup_workflow.py

```diff
@@ -1334,19 +1516,41 @@ def create_xml(dict_configs):
         dict_hyb_tasks = get_hyb_tasks(dict_configs)
 
         # Removes <memory>&MEMORY_JOB_DUMP</memory> post mortem from hyb tasks
-        hyp_tasks = {'gdaseobs':'gdaseobs',
-                     'gdasediag':'gdasediag',
-                     'gdaseomg':'gdaseomn',
-                     'gdaseupd':'gdaseupd',
-                     'gdasecen':'gdasecmn',
-                     'gdasesfc':'gdasesfc',
-                     'gdasefcs':'gdasefmn',
-                     'gdasepos':'gdasepmn',
-                     'gdasearc':'gdaseamn',
-                     'gdasechgres':'gdasechgres'}
+        do_efsoi = base.get('DO_EFSOI', 'NO').upper()
+
+        if do_efsoi in ['Y', 'YES']:
+            hyp_tasks = {'gdaseobs':'gdaseobs',
+                         'gdasediag':'gdasediag',
+                         'gdaseomg':'gdaseomn',
+                         'gdaseupd':'gdaseupd',
+                         'gdasecen':'gdasecmn',
+                         'gdasesfc':'gdasesfc',
+                         'gdasefcs':'gdasefmn',
+                         'gdasepos':'gdasepmn',
+                         'gdaseupdfsoi':'gdaseupdfsoi',
+                         'gdasecenfsoi':'gdasecmnfsoi',
+                         'gdasesfcfsoi':'gdasesfcfsoi',
+                         'gdasefcsfsoi':'gdasefmnfsoi',
+                         'gdaseposfsoi':'gdasepmnfsoi',
+                         'gdasefsoi':'gdasefsoi',
+                         'gdasearc':'gdaseamn',
+                         'gdasechgres':'gdasechgres'}
+
+        else:
+            hyp_tasks = {'gdaseobs':'gdaseobs',
+                         'gdasediag':'gdasediag',
+                         'gdaseomg':'gdaseomn',
+                         'gdaseupd':'gdaseupd',
+                         'gdasecen':'gdasecmn',
+                         'gdasesfc':'gdasesfc',
+                         'gdasefcs':'gdasefmn',
+                         'gdasepos':'gdasepmn',
+                         'gdasearc':'gdaseamn',
+                         'gdasechgres':'gdasechgres'}
+        #for each_task, each_resource_string in dict_hyb_resources.iteritems():
         for each_task, each_resource_string in dict_hyb_resources.items():
-            #print(each_task,hyp_tasks[each_task])
-            #print(dict_hyb_tasks[hyp_tasks[each_task]])
+            #print each_task,hyp_tasks[each_task]
```

Comment:
The removal of the parentheses would makes these lines not work if uncommented, but they should probably just be removed. If they remain for debugging, they should be modified so we aren't defining whole dictionaries just to serve this debug print.

Reply:
removed